File: Contest/CodeForce/170406/E/main.py
import sys
s = raw_input()
m = input()
k = 1;
l = len(s)
for i in range(1,l-1) :
    k = (10*k)%m;
r = int(s[0])
for i in range(1,l-1) :
    r = ((r*10)+int(s[i]))%m;
res = r;
for i in range(0,l-2) :
    frt = int(s[i])
    r = (r + m - (k*frt)%m)%m;
    r = ((r*10)+(frt))%m;
    if int(s[i+1]) > 0 and res > r:
         res = r;
print(res)

File: Contest/CodeForce/170406/E/main.cpp
#include <iostream>
#include <cstring>
using namespace std;
int main() {

    string str;
    cin >> str;
    int m;
    cin >> m;

    int k = 1;
    for (int i=1; i<str.size(); i++)
        k = (10*k)%m;

    long long r = str[0]-'0';
    for (int i=1; i<str.size(); i++) {
        r = ((r*10)+(str[i]-'0'))%m;
    }
    long long res = r;
    for (int i=0; i<str.size()-1; i++) {
        int frt = str[i]-'0';
        r = (r + m - (k*frt)%m)%m;
        r = ((r*10)+(str[i]-'0'))%m;
        if (str[i+1] > 0 && res > r)
            res = r;
    }
    cout << res << endl;
}
